keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -71,255 +76,298 @@ private void verifyEnvironmentVariables() {
+    createTestSecret(true, false);
+    createTestSecret(false, true);
+    createTestSecret(true, true);
+  }
+
+  private static void deleteTestSecrets() {
+    testSecretNames.forEach(
+        name -> {
+          final DeleteSecretRequest deleteSecretRequest =
+              DeleteSecretRequest.builder().secretId(name).build();
+          testSecretsManagerClient.deleteSecret(deleteSecretRequest).join();
+        });
+    testSecretNames.clear();
+    allTestSecretTags.clear();
+    testSecretSingleTag.clear();
+    testSecretMultipleTags.clear();
+    testSecretSingleSharedTag.clear();
+    testSecretMultipleSharedTags.clear();
+  }
+
+  private static Tag createTag(final String key, final String value) {
+    return Tag.builder().key(key).value(value).build();
+  }
+
+  private static List<Tag> createTestSecretTags(
+      final String secretName, final boolean hasMultipleTags, final boolean hasSharedTags) {
+    final List<Tag> testSecretTags = new ArrayList<>();
+    testSecretTags.add(createTag(secretName, secretName));
+    if (hasMultipleTags) {
+      testSecretTags.add(createTag(secretName + "/multiple", "multiple"));
+    }
+    if (hasSharedTags) {
+      allTestSecretTags.forEach((key, value) -> testSecretTags.add(createTag(key, "shared")));
+    }
+    return testSecretTags;
+  }
+
+  private static void updateTestTags(
+      final List<Tag> tags, final boolean multipleTags, final boolean sharedTags) {
+    tags.forEach(
+        tag -> {
+          if (!multipleTags && !sharedTags) {
+            testSecretSingleTag.put(tag.key(), tag.value());
+          } else if (multipleTags && sharedTags) {
+            testSecretMultipleTags.put(tag.key(), tag.value());
+          } else if (!multipleTags) {
+            testSecretSingleSharedTag.put(tag.key(), tag.value());
+          } else {
+            testSecretMultipleSharedTags.put(tag.key(), tag.value());
+          }
+        });
+    allTestSecretTags = new HashMap<>();
+    allTestSecretTags.putAll(testSecretSingleTag);
+    allTestSecretTags.putAll(testSecretMultipleTags);
+    allTestSecretTags.putAll(testSecretSingleSharedTag);
+    allTestSecretTags.putAll(testSecretMultipleSharedTags);
+  }
+
+  private static void waitUntilSecretAvailable(final String secretName) {
+    testSecretsManagerClient
+        .getSecretValue(GetSecretValueRequest.builder().secretId(secretName).build())
+        .join();
+  }
+
+  private void validateMappedSecret(
+      final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries,
+      final Map<String, String> testTags) {
+    testTags.keySet().stream()
+        .filter(tagKey -> testSecretNames.contains(tagKey))
+        .collect(Collectors.toList())
+        .forEach(
+            tagKey ->
+                assertThat(tagKey)
+                    .isIn(
+                        secretEntries.stream()
+                            .map(entry -> entry.getKey())
+                            .collect(Collectors.toList())));
```

Comment:
```suggestion
    testTags.keySet().stream()
        .filter(tagKey -> testSecretNames.contains(tagKey))
        .forEach(tagKey ->
            assertThat(tagKey).isIn(secretEntries.stream().map(e -> e.getKey()))
        );
```

This would remove the unnecessary list conversions.